### PR TITLE
Commit kafka offset in routine load

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Commit kafka offset in routine load

Kafka will decide whether to delete data based on whether all consumer group is commit offset or not. 
If there is no commit offset, the kafka server disk may be full

### CODE REVIEW

File: be/src/runtime/routine_load/routine_load_task_executor.cpp [hunk 1]
```diff
@@ -227,6 +227,28 @@ void RoutineLoadTaskExecutor::exec_task(
...
+    // commit kafka offset
+    switch (ctx->load_src_type) {
+        case TLoadSourceType::KAFKA: {
+            std::shared_ptr<DataConsumer> consumer;
+            HANDLE_ERROR(_data_consumer_pool.get_consumer(ctx, &consumer), "failed to get consumers");
+            std::vector<RdKafka::TopicPartition*> topic_partitions;
+
+            for (auto& kv : ctx->kafka_info->cmt_offset) {
+                RdKafka::TopicPartition* tp1 = RdKafka::TopicPartition::create(
+                        ctx->kafka_info->topic, kv.first, kv.second);
```
Comment (morningman) [inline be/src/runtime/routine_load/routine_load_task_executor.cpp]: Do we need to delete this `tp1` after commit?

File: be/src/runtime/routine_load/routine_load_task_executor.cpp [hunk 2]
```diff
@@ -227,6 +227,28 @@ void RoutineLoadTaskExecutor::exec_task(
     // commit txn
     HANDLE_ERROR(_exec_env->stream_load_executor()->commit_txn(ctx), "commit failed");
 
+    // commit kafka offset
+    switch (ctx->load_src_type) {
+        case TLoadSourceType::KAFKA: {
+            std::shared_ptr<DataConsumer> consumer;
+            HANDLE_ERROR(_data_consumer_pool.get_consumer(ctx, &consumer), "failed to get consumers");
```
Comment (morningman) [inline be/src/runtime/routine_load/routine_load_task_executor.cpp]: It's better not return any error if we failed to do something after txn is committed.

**Review by morningman — CHANGES_REQUESTED** [review]

File: be/src/runtime/routine_load/data_consumer.h [hunk 3]
```diff
@@ -133,6 +133,8 @@ class KafkaDataConsumer : public DataConsumer {
     // reassign partition topics
     virtual Status reset() override;
     virtual bool match(StreamLoadContext* ctx) override;
+    // commit kafka offset
+    Status commit(std::vector<RdKafka::TopicPartition*> offset);
```
Comment (imay) [inline be/src/runtime/routine_load/data_consumer.h]:
```suggestion
    Status commit(const std::vector<RdKafka::TopicPartition*>& offset);
```

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**